pkg/scalers/pulsar_scaler.go

```diff
@@ -0,0 +1,313 @@
+		if err != nil {
+			return meta, fmt.Errorf("error parsing %s: %s", msgBacklogMetricName, err)
+		}
+		meta.msgBacklogThreshold = t
+	}
+
+	meta.enableTLS = false
+	if val, ok := config.TriggerMetadata["tls"]; ok {
+		val = strings.TrimSpace(val)
+
+		if val == "enable" {
+			cert := config.AuthParams["cert"]
+			key := config.AuthParams["key"]
+			if key == "" || cert == "" {
+				return meta, errors.New("must be provided cert and key")
+			}
+			meta.ca = config.AuthParams["ca"]
+			meta.cert = cert
+			meta.key = key
+			meta.enableTLS = true
+		} else {
+			return meta, fmt.Errorf("err incorrect value for TLS given: %s", val)
+		}
+	}
+	return meta, nil
+}
+
+func (s *pulsarScaler) GetStats() (*pulsarStats, error) {
+	stats := new(pulsarStats)
+
+	topic := strings.ReplaceAll(s.metadata.topic, "persistent://", "")
+	statsURL := s.metadata.adminURL + "/admin/v2/persistent/" + topic + "/stats"
+
+	req, err := http.NewRequest("GET", statsURL, nil)
+	if err != nil {
+		pulsarLog.Error(err, "error requesting stats from url")
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	res, err := s.client.Do(req)
+	if res == nil || err != nil {
+		pulsarLog.Error(err, "error requesting stats from url")
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case 200:
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			pulsarLog.Error(err, "error requesting stats from url")
+			return nil, fmt.Errorf("error requesting stats from url: %s", err)
+		}
+		err = json.Unmarshal(body, stats)
+		if err != nil {
+			pulsarLog.Error(err, "error unmarshalling response")
+			return nil, fmt.Errorf("error unmarshalling response: %s", err)
+		}
+		return stats, nil
+	case 404:
+		return nil, nil
+	default:
+		pulsarLog.Error(err, "error requesting stats from url")
+		return nil, fmt.Errorf("error requesting stats from url: %s", err)
+	}
+}
+
+func (s *pulsarScaler) getMsgBackLog() (int, bool, error) {
+	stats, err := s.GetStats()
+	if err != nil {
+		return 0, false, err
+	}
+
+	if stats == nil {
+		return 0, false, nil
+	}
+
+	v, found := stats.Subscriptions[s.metadata.subscription]
+
+	return v.Msgbacklog, found, nil
+}
+
+// IsActive determines if we need to scale from zero
+func (s *pulsarScaler) IsActive(_ context.Context) (bool, error) {
+	msgBackLog, found, err := s.getMsgBackLog()
+	if err != nil {
+		return false, err
+	}
+
+	if !found || msgBackLog == 0 {
+		pulsarLog.Info("Pulsar subscription is not active, either no subscription found or no backlog detected")
+		return false, nil
+	}
+
+	pulsarLog.Info("Pulsar subscription is active with backlog")
+	return true, nil
+}
+
+// GetMetrics returns value for a supported metric and an error if there is a problem getting the metric
+func (s *pulsarScaler) GetMetrics(_ context.Context, metricName string, _ labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
```

Comment:
Please, use this context and propagate it to http request using [http.NewRequestWithContext](https://pkg.go.dev/net/http#NewRequestWithContext)